x-pack/plugins/infra/public/alerting/inventory/components/expression.tsx

```diff
@@ -257,7 +257,7 @@ export const Expressions: React.FC<Props> = (props) => {
       preFillAlertCriteria();
     }
 
-    if (!ruleParams.filterQuery) {
+    if (ruleParams.filterQuery === undefined) {
```

Comment:
The reason for the bug with removing the prefill was that once the clear option was used the `ruleParams.filterQuery`  was set to an empty string which triggered the prefill again.